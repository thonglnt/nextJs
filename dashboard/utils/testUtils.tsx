import { ReactNode } from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from '@mui/material/styles'

// Themes
import theme from '@self-types/Themes.types'

export const customRender = (children: ReactNode) => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider>)
}

export * from '@testing-library/react'
export { customRender as render }
