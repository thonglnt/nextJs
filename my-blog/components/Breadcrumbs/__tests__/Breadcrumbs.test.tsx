import { cleanup, fireEvent, render } from '@utils/testUtils'
import BasicBreadcrumbs, { IBreadcrumbs } from '..'

describe('Breadcrumbs component', () => {
  afterEach(cleanup)

  const mockProps: IBreadcrumbs = {
    links: [
      {
        link: 'Blog',
        href: '/blog',
      },
    ],
  }

  it('should render Breadcrumb component', () => {
    const { getByText } = render(<BasicBreadcrumbs {...mockProps} />)

    expect(getByText).toMatchSnapshot()

    const link = getByText('HOME')
    expect(link).toBeTruthy()

    expect(link).toHaveAttribute('href', '/')
  })

  it('should redirect corresponding url when click on breadcrumb link', () => {
    const { container } = render(<BasicBreadcrumbs {...mockProps} />)

    fireEvent.click(container)

    expect(window.location.href).toBe('http://localhost/')
  })
})
