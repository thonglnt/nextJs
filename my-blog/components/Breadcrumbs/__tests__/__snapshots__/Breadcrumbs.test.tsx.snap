// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs component should render Breadcrumb component 1`] = `[Function]`;
